adapters/utils/utils.ts

```diff
@@ -0,0 +1,54 @@
+import { Balances } from "@defillama/sdk"
+import { FetchResponseValue } from "../types";
+
+export function createBalanceFrom(chain: string, timestamp: number | undefined, amount: FetchResponseValue | undefined): Balances {
+  const balance = new Balances({ chain, timestamp })
+  if (amount) {
+    if (typeof amount === 'number' || typeof amount === 'string') {
+      balance.addUSDValue(amount)
+    } else {
+      balance.addBalances(amount)
+    }
+  }
+  return balance;
+}
+
+export async function subtractBalance(balance: Balances, amount: FetchResponseValue | undefined) {
+  if (amount) {
+    if (typeof amount === 'number' || typeof amount === 'string') {
+      const usdValue = await balance.getUSDValue()
+      balance.resizeBy(0)
```

Comment:
this is a bit messy, you are essentially resetting balances object, and we lose token breakdown now? why not do this as 

    const otherBalance = createBalanceFrom(balance.chain, balance.timestamp, amount)
    balance.subtract(otherBalance)